lib/node_modules/@stdlib/string/base/replace-before/README.md

```diff
@@ -0,0 +1,233 @@
+<!--
+
+@license Apache-2.0
+
+Copyright (c) 2023 The Stdlib Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# replaceBefore
+
+> Replace the substring before the first occurrence of a specified search string.
+
+<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->
+
+<section class="intro">
+
+</section>
+
+<!-- /.intro -->
+
+<!-- Package usage documentation. -->
+
+<section class="usage">
+
+## Usage
+
+```javascript
+var replaceBefore = require( '@stdlib/string/base/replace-before' );
+```
+
+#### replaceBefore( str, search, replacement )
+
+Replaces the substring before the first occurrence of a specified search string.
+
+```javascript
+var str = 'beep boop';
+var replacement = 'loop';
+var out = replaceBefore( str, ' ', replacement );
+// returns 'loop boop'
+
+replacement = 'bar';
+out = replaceBefore( str, 'o' , replacement );
+// returns 'baroop'
+```
+
+</section>
+
+<!-- /.usage -->
+
+<!-- Package usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="notes">
+
+## Notes
+
+-   If a substring is not present in a provided string, the function returns the input string.
+-   If provided an empty substring, the function returns an empty string.
+
+</section>
+
+<!-- /.notes -->
+
+<!-- Package usage examples. -->
+
+<section class="examples">
+
+## Examples
+
+<!-- eslint no-undef: "error" -->
+
+```javascript
+var substringBefore = require( '@stdlib/string/base/replace-before' );
+
+var out = substringBefore( 'beep boop', 'p', 'see' );
+// returns 'seep boop'
+
+out = substringBefore( 'Hello World!', 'xyz', 'foo' );
+// returns 'fooHello World!'
```

Comment:
This looks wrong to me. If the search string is not present, we should return the input string unchanged.